sc/src/main/java/uk/ac/ebi/atlas/tsne/TSnePoint.java

```diff
@@ -26,7 +26,7 @@ public int compareTo(@Nullable TSnePoint o) {
         return o == null ? -1 : this.name().compareTo(o.name());
     }
 
-    // The public static method returning a TypeAdapter<Foo> is what tells auto-value-gson to create a TypeAdapter for Foo.
+    // Public static method returning a TypeAdapter<Foo> is what tells auto-value-gson to create a TypeAdapter for Foo
```

Comment:
If it works it's fine - looks fancy, and I'm not sure what it gives you. I just create toJson() methods everywhere.

Reply:
It’s a very simple way to add Gson (de)serializing support in `@AutoValue` classes. More here: https://github.com/rharter/auto-value-gson